velox/vector/VariantToVector.cpp

```diff
@@ -89,6 +111,12 @@ ArrayVectorPtr variantArrayToVector(
   }
 
   const auto elementType = variantArray.at(nonNullIndex.value()).inferType();
+  if (elementType->kind() == TypeKind::SHORT_DECIMAL) {
```

Comment:
this specialization is not required since you are using the same `variantArrayToVectorImpl` function.

Reply:
Thanks for the suggestion but removing this specialization results in a compilation error:
```
velox/velox/vector/VariantToVector.cpp:66:31: error: no matching conversion for functional-style cast from 'const facebook::velox::detail::OpaqueCapsule' to 'T' (aka 'shared_ptr<void>')
        arrayElements->set(i, T(variantArray[i].value<KIND>()));
```

Reply:
Are you using VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL?
May be specialize. 
```
ArrayVectorPtr variantArrayToVectorImpl<TypeKind::OPAQUE>(
    const std::vector<variant>& variantArray,
    velox::memory::MemoryPool* pool,
    const TypePtr& type) {
    VELOX_NYI();
}
```

Reply:
Yes, using VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL still resulted in this error. Specialized for OPAQUE as suggested, thanks!